src/hotspot/share/classfile/systemDictionary.cpp

```diff
@@ -1526,6 +1526,10 @@ InstanceKlass* SystemDictionary::find_or_define_instance_class(Symbol* class_nam
   } else if (HAS_PENDING_EXCEPTION) {
     assert(defined_k == nullptr, "Should not have a klass if there's an exception");
     k->class_loader_data()->add_to_deallocate_list(k);
+
+    // Also remove this InstanceKlass from the LoaderConstraintTable if added.
```

Comment:
Why do we not do this in the non-exception case? If k wasa duplicate case then there should be a constraint entry under `defined_k`, so any entry under `k` should also be removed shouldn't it?

Reply:
Okay